nilearn/connectivity/connectivity_matrices.py

```diff
@@ -0,0 +1,349 @@
+                      "getting to the requested tolerance level "
+                      "{1}.".format(max_iter, tol))
+
+    return gmean
+
+
+def sym_to_vec(symmetric):
+    """Return the flattened lower triangular part of an array, after
+    multiplying above the diagonal elements by sqrt(2).
+
+    Acts on the last two dimensions of the array if not 2-dimensional.
+
+    Parameters
+    ----------
+    symmetric : numpy.ndarray, shape (..., n_features, n_features)
+        Input array.
+
+    Returns
+    -------
+    output : numpy.ndarray, shape (..., n_features * (n_features + 1) / 2)
+        The output flattened lower triangular part of symmetric.
+    """
+    scaling = sqrt(2) * np.ones(symmetric.shape[-2:])
+    np.fill_diagonal(scaling, 1.)
+    tril_mask = np.tril(np.ones(symmetric.shape[-2:])).astype(np.bool)
+    return symmetric[..., tril_mask] * scaling[tril_mask]
+
+
+def _cov_to_corr(covariance):
+    """Return correlation matrix for a given covariance matrix.
+
+    Parameters
+    ----------
+    covariance : 2D numpy.ndarray
+        The input covariance matrix.
+
+    Returns
+    -------
+    correlation : 2D numpy.ndarray
+        The ouput correlation matrix.
+    """
+    diagonal = np.atleast_2d(1. / np.sqrt(np.diag(covariance)))
+    correlation = covariance * diagonal * diagonal.T
+    return correlation
+
+
+def _prec_to_partial(precision):
+    """Return partial correlation matrix for a given precision matrix.
+
+    Parameters
+    ----------
+    precision : 2D numpy.ndarray
+        The input precision matrix.
+
+    Returns
+    -------
+    partial_correlation : 2D numpy.ndarray
+        The 2D ouput partial correlation matrix.
+    """
+    partial_correlation = -_cov_to_corr(precision)
+    np.fill_diagonal(partial_correlation, 1.)
+    return partial_correlation
+
+
+class ConnectivityMeasure(BaseEstimator, TransformerMixin):
+    """A class that computes different kinds of functional connectivity
+    matrices.
+
+    Parameters
+    ----------
+    cov_estimator : estimator object, optional.
+        The covariance estimator.
+
+    kind : {"correlation", "partial correlation", "robust dispersion",\
+            "covariance", "precision"}, optional
+        The matrix kind.
+
+    Attributes
+    ----------
+    `cov_estimator_` : estimator object
+        A new covariance estimator with the same parameters as cov_estimator.
+
+    `robust_mean_` : numpy.ndarray
+        The mean connectivity for the robust dispersion kind.
+
+    `whitening_` : numpy.ndarray
+        The inverted square-rooted geometric mean of the covariance matrices.
+
+    References
+    ----------
+    For the use of "robust dispersion", see the paper:
+    G. Varoquaux et al. "Detection of brain functional-connectivity difference
+    in post-stroke patients using group-level covariance modeling, MICCAI 2010.
+    """
+
+    def __init__(self, cov_estimator=EmpiricalCovariance(assume_centered=True),
```

Comment:
> >     Any reason that we are not using a LedoitWolf by default?

>     for a bad reason: I always observed that 'tangent' gives better results
>     with EmpiricalCovariance

You mean that it makes tangent space look better compared to the others?
Here, we are interested in getting overall good performance in every
scenario. The defaults in nilearn should be as robust as possible.